File: src/main/java/com/boardgame/reservation/member/domain/Role.java
package com.boardgame.reservation.member.domain;

/**
 * 회원 권한. Spring Security의 hasRole("ADMIN")은 내부적으로 "ROLE_ADMIN" 권한을 확인한다.
 * 계층(SecurityConfig.roleHierarchy): SUPER_ADMIN > ADMIN > USER
 * SUPER_ADMIN 은 부트스트랩(ADMIN_EMAIL) 계정 1개뿐이며 가입·승인으로는 만들 수 없다.
 */
public enum Role {
    USER,
    ADMIN,
    SUPER_ADMIN;

    public String getAuthority() {
        return "ROLE_" + name();
    }
}

File: frontend/src/utils/role.ts
import type { Role } from '../types/auth.ts'

// 백엔드 RoleHierarchy: SUPER_ADMIN > ADMIN > USER. hasRole('ADMIN') API 는 SUPER_ADMIN 도 통과한다.
// role === 'ADMIN' 처럼 직접 비교하면 SUPER_ADMIN 이 빠지므로 화면에서는 이 함수를 쓴다. (실제 방어는 백엔드 403)

export function isAdmin(role: Role): boolean {
  return role === 'ADMIN' || role === 'SUPER_ADMIN'
}

export function isSuperAdmin(role: Role): boolean {
  return role === 'SUPER_ADMIN'
}
